tests/contrib/dramatiq/test_snapshots.py

```diff
@@ -0,0 +1,102 @@
+import unittest
+
+import dramatiq
+
+from ddtrace.contrib.dramatiq import patch
+from ddtrace.contrib.dramatiq import unpatch
+from tests.utils import snapshot
+
+
+class DramatiqSnapshotTests(unittest.TestCase):
+    def setUp(self):
+        patch()
+
+    def tearDown(self):
+        unpatch()
+
+    @snapshot(wait_for_num_traces=2)
+    def test_idempotent_patch(self):
+        # calling patch() twice doesn't have side effects
+        patch()
+
+        @dramatiq.actor
+        def fn_task():
+            return "hi"
+
+        fn_task.send()
+        fn_task.send_with_options(options={"max_retries": 1})
+
+    @snapshot(wait_for_num_traces=0)
```

Comment:
This and line 42 for `test_fn_task_synchronous()`. This isn't actually doing what I want it to do, which is to confirm that there are 0 traces generated. It looks like `wait_for_num_traces` could accidentally prematurely consider this correct by meeting the loop exit condition, when in reality the call did generate a trace. Leaving this blank causes an indexing error though, so maybe this IS ok? Doesn't feel right though, so looking for suggestions on making this better.

Reply:
If you're wanting to check that no traces are generated, a snapshot test might not be the best fit. Instead, you could set up a `DummyTracer` and then verify that its list of traces was empty, similarly to [this example](https://github.com/DataDog/dd-trace-py/blob/fedf88e7161e7eff19038231c2b2fa8f8f75ddec/tests/contrib/cassandra/test.py#L441-L449).

Reply:
`Pin`!! 📌